addons/pkg-maven/common/src/main/java/org/commonjava/indy/pkg/maven/content/MavenMetadataGenerator.java

```diff
@@ -366,21 +377,20 @@ public Transfer generateGroupFileContent( final Group group, final List<Artifact
         if ( exists( target ) )
         {
             // Means there is no metadata change if this transfer exists, so directly return it.
-            logger.trace( "Metadata file exists for group {} of path {}, no need to regenerate.", group.getKey(), path );
+            logger.trace( "Metadata file exists for group {} of path {}, no need to regenerate.", group.getKey(),
+                          path );
             return target;
         }
 
-        final ReentrantLock mergerLock = getMergerLock( group, toMergePath );
-        final boolean locked = mergerLock.tryLock();
-        boolean mergingDone = false;
-
-        if ( locked )
-        {
+        AtomicReference<IndyWorkflowException> wfEx = new AtomicReference<>();
+        String mergePath = toMergePath;
+        boolean mergingDone = mergerLocks.ifUnlocked( computeKey(group, toMergePath), p->{
```

Comment:
What is the meaning of "ifunlocked"? It seems the lockedFunction just do a tryLock() and returns true/false and "ifunlocked" does not do anything further based on the result and just returns -- refer to "return lockedFunction.apply( key, lock );". The lockAnd() is easy to understand. Why not use it here?